sql/expression/function/convert_tz.go

```diff
@@ -0,0 +1,197 @@
+// Copyright 2021 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package function
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/dolthub/go-mysql-server/sql"
+)
+
+var offsetRegex = regexp.MustCompile(`(?m)^\+(\d{2}):(\d{2})$`)
+
+type ConvertTz struct {
+	dt     sql.Expression
+	fromTz sql.Expression
+	toTz   sql.Expression
+}
+
+var _ sql.FunctionExpression = (*ConvertTz)(nil)
+
+// NewConvertTz returns an implementation of the CONVERT_TZ() function.
+func NewConvertTz(ctx *sql.Context, dt, fromTz, toTz sql.Expression) sql.Expression {
+	return &ConvertTz{
+		dt:     dt,
+		fromTz: fromTz,
+		toTz:   toTz,
+	}
+}
+
+// Resolved implements the sql.Expression interface.
+func (c *ConvertTz) Resolved() bool {
+	return c.dt.Resolved() && c.fromTz.Resolved() && c.toTz.Resolved()
+}
+
+// String implements the sql.Expression interface.
+func (c *ConvertTz) String() string {
+	return fmt.Sprintf("CONVERT_TZ(%s, %s, %s)", c.dt, c.fromTz, c.toTz)
+}
+
+// Type implements the sql.Expression interface.
+func (c *ConvertTz) Type() sql.Type {
+	return sql.Datetime
+}
+
+// IsNullable implements the sql.Expression interface.
+func (c *ConvertTz) IsNullable() bool {
+	return true
+}
+
+// Eval implements the sql.Expression interface.
+func (c *ConvertTz) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
+	from, err := c.fromTz.Eval(ctx, row)
+	if err != nil {
+		return nil, err
+	}
+
+	to, err := c.toTz.Eval(ctx, row)
+	if err != nil {
+		return nil, err
+	}
+
+	datetime, err := c.dt.Eval(ctx, row)
+	if err != nil {
+		return nil, err
+	}
+
+	datetimeStr, ok := datetime.(string)
```

Comment:
This isn't right. This value can be anything that can be converted to a datetime, not just a string

Reply:
Need a test for this